e2e/test/ServiceClientE2ETests.cs

```diff
@@ -14,6 +15,9 @@ namespace Microsoft.Azure.Devices.E2ETests
     [TestCategory("IoTHub-E2E")]
     public class ServiceClientE2ETests : IDisposable
     {
+        private readonly string DirectMethodName = $"{nameof(ServiceClientE2ETests)}Method";
+        private TimeSpan CustomTimeout = TimeSpan.FromMinutes(5);
```

Comment:
I redid these e2e tests, but we do still need a long timeout in one of these tests because that is the only way to exercise the code path that this PR is bringing proxy support to. Any method invocation with a read timeout + connect timeout that is less than the default operation timeout will use the default operation timeout code path.